WordPress/src/main/java/org/wordpress/android/ui/posts/TagsFragment.java

```diff
@@ -0,0 +1,190 @@
+package org.wordpress.android.ui.posts;
+
+import android.os.Bundle;
+import android.text.Editable;
+import android.text.TextUtils;
+import android.text.TextWatcher;
+import android.view.KeyEvent;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.EditText;
+
+import androidx.annotation.LayoutRes;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+import org.apache.commons.text.StringEscapeUtils;
+import org.greenrobot.eventbus.Subscribe;
+import org.greenrobot.eventbus.ThreadMode;
+import org.wordpress.android.R;
+import org.wordpress.android.WordPress;
+import org.wordpress.android.fluxc.Dispatcher;
+import org.wordpress.android.fluxc.model.SiteModel;
+import org.wordpress.android.fluxc.store.TaxonomyStore;
+import org.wordpress.android.fluxc.store.TaxonomyStore.OnTaxonomyChanged;
+import org.wordpress.android.util.ActivityUtils;
+
+import javax.inject.Inject;
+
+import static org.wordpress.android.ui.posts.PostSettingsTagsActivity.KEY_TAGS;
+
+public abstract class TagsFragment extends Fragment implements TextWatcher, View.OnKeyListener, TagSelectedListener {
```

Comment:
@ashiagr I ended up having to create an abstract class so that I can override the layout and `onAttach` in the respective fragments that extend this.